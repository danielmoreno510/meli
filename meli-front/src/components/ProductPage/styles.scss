@import '../../index';

.product-container {
  display: flex;
  justify-content: space-between;
}

.product-image {
  width: 400px;
}

.product-section {
  width: 300px;
}

.product-condition {
  font-size: 22px;
}

.product-title {
  font-size: 26px;
  font-weight: 500;
  margin: 0px;
  margin-top: 6px;
}

.product-amount {
  font-size: 42px;
  font-weight: 300;
  margin-top: 12px;
}

.product-button {
  width: 100%;
  height: 56px;
  background-color: $button-color;
  border-radius: 6px;
  color: $layout-color;
  font-size: 32px;
  text-align: center;
  padding-top: 8px;
  margin-top: 32px;
}

.product-description-title{
  font-size: 32px;
  font-weight: 400;
  margin: 0px;
  margin-top: 80px;
}

.product-description {
  margin-top: 20px;
  width: 60%;
}