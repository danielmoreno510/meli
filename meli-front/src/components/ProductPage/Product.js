import PropTypes from "prop-types";

import { formatCurrency } from "../../utils/currencyUtils";
import "./styles.scss";

const Product = ({ product }) => (
  <div>
    <div className="product-container">
      <img
        className="product-image"
        src={product.picture}
        alt={product.title}
      />
      <div className="product-section">
        <div className="product-condition">
          {product.condition} - {product.sold_quantity} vendidos
        </div>
        <h1 className="product-title">{product.title}</h1>
        <div className="product-amount">
          {formatCurrency(product.price.currency, product.price.amount)}
        </div>
        <div className="product-button">Comprar</div>
      </div>
    </div>
    <h2 className="product-description-title">Descripcion del producto</h2>
    <div className="product-description">{product.description}</div>
  </div>
);

Product.propTypes = {
    product: PropTypes.object,
  };

  export default Product