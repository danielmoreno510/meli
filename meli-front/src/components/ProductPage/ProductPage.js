import { useEffect } from "react";
import { useLocation } from "react-router-dom";
import PropTypes from "prop-types";

import WithLoadingIndicator from "../WithLoading";
import Product from "./Product"
import "./styles.scss";

const ProductWithLoadingIndicator = WithLoadingIndicator(Product);

const ProductPage = ({ selectedProduct, product, isFetchingProduct, getProduct }) => {
  const { pathname } = useLocation();
  const id = pathname.replace("/items/", "");

  useEffect(() => {
    !selectedProduct && getProduct(id);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [id, getProduct]);

  return (
    <ProductWithLoadingIndicator
      product={product}
      isFetching={isFetchingProduct || !product.id}
    />
  );
};

ProductPage.propTypes = {
  selectedProduct: PropTypes.bool,
  product: PropTypes.object,
  isFetchingProduct: PropTypes.bool,
  getProduct: PropTypes.func
};

export default ProductPage;
