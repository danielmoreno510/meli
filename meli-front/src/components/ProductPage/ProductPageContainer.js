import ProductPage from "./ProductPage";
import { fetchProduct } from "../../state/actions";
import { StoreService } from "../../state/StoreService.js";

export const mapStateToProps = (state) => ({
  selectedProduct: state.products.selectedProduct,
  product: state.products.product,
  isFetchingProduct: state.products.isFetchingProduct
});
export const mapDispatchToProps = (dispatch) => {
  return {
    getProduct: (id) => {
      dispatch(fetchProduct(id));
    }
  };
};
const ProductPageContainer = StoreService.connect(
  mapStateToProps,
  mapDispatchToProps
)(ProductPage);
export default ProductPageContainer;
