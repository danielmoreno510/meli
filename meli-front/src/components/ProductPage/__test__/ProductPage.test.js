import React from "react";
import { shallow } from "enzyme";

import ProductPage from "../ProductPage";

jest.mock("react-router-dom", () => {
  return {
    useLocation: jest.fn().mockReturnValue({ pathname: "/" }),
    useHistory: () => ({
      push: jest.fn(),
    }),
  };
});

describe("ProductPage component", () => {
  const testProps = {
    isFetchingProduct: false,
    product: {
      id: "MLA916044386",
      thumbnail: "image.png",
      title: "iPhone 11 64 Gb Blanco",
      price: { currency: "ARS", amount: "123456" },
    },
    getProduct: jest.fn(),
  };

  describe("WHEN the ProductPage component is render", () => {
    const wrapper = shallow(<ProductPage {...testProps} />);

    it("THEN should display ProductPage component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    const product = wrapper
      .find("ComponentWithLoadingIndicator")
      .dive()
      .find("Product");

    it("THEN should exist the ProductPage component", () => {
      expect(product.exists()).toBeTruthy();
    });
  });
});
