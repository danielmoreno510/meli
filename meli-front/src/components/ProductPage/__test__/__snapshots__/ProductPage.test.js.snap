// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPage component WHEN the ProductPage component is render THEN should display ProductPage component 1`] = `
<ComponentWithLoadingIndicator
  isFetching={false}
  product={
    Object {
      "id": "MLA916044386",
      "price": Object {
        "amount": "123456",
        "currency": "ARS",
      },
      "thumbnail": "image.png",
      "title": "iPhone 11 64 Gb Blanco",
    }
  }
/>
`;
